onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -489,12 +489,18 @@ common::Status TensorrtExecutionProvider::Compile(const std::vector<onnxruntime:
         int output_index = output_indexes[i];
         output_shapes[i].insert(output_shapes[i].begin(), batch_size);
 
-        // If output tensor type is INT64, TensorRT processes data as INT32 and the output will be converted to INT64.
+        int output_size = batch_size * output_dim_sizes[i];
         OrtValue* output_tensor = ort.KernelContext_GetOutput(context, output_index, output_shapes[i].data(), output_shapes[i].size());
-        if (output_types[i] == TensorProto::FLOAT) {        
-          CHECK_CUDA(cudaMemcpy(ort.GetTensorMutableData<float>(output_tensor), buffers[i + num_binding_inputs], batch_size * output_dim_sizes[i] * sizeof(float), cudaMemcpyDeviceToHost));
+        // If output tensor type is INT64, TensorRT processes data as INT32 and the output will be converted to INT64.
+        if (output_types[i] == TensorProto::FLOAT) {
+          CHECK_CUDA(cudaMemcpy(ort.GetTensorMutableData<float>(output_tensor), buffers[i + num_binding_inputs], output_size * sizeof(float), cudaMemcpyDeviceToHost));
         } else if (output_types[i] == TensorProto::INT64) {
```

Comment:
shouldn't there be a TensorProto:INT32 case?

Reply:
So far all the models have either FLOAT or INT64 outputs. I plan to add other data types that Nvidia claimed to support such as INT8, FLOAT16 and INT32 in another PR once I tested them in the models.

Reply:
doesn't seem safe to go by the handful of models you manually tested.
what about the case of intermediate outputs (for partitioned subgraphs)
it seems safer add the types that are supported by TensorRT and supported in ONNX.
otherwise, your else case "output type not supported by TensorRT" is not really true and may cause confusion.